lib/Pico.php

```diff
@@ -0,0 +1,945 @@
+
+        // replace %meta.*%
+        $metaKeys = array_map(function ($metaKey) {
+            return '%meta.' . $metaKey . '%';
+        }, array_keys($this->meta));
+        $metaValues = array_values($this->meta);
+        $content = str_replace($metaKeys, $metaValues, $content);
+
+        return $content;
+    }
+
+    /**
+     * Parses the contents of a page using ParsedownExtra
+     *
+     * @param  string $content raw contents of a page (Markdown)
+     * @return string          parsed contents (HTML)
+     */
+    public function parseFileContent($content)
+    {
+        $parsedown = new ParsedownExtra();
+        return $parsedown->text($content);
+    }
+
+    /**
+     * Returns the cached contents of the requested page
+     *
+     * @see    Pico::parseFileContent()
+     * @return string parsed contents
+     */
+    public function getFileContent()
+    {
+        return $this->content;
+    }
+
+    /**
+     * Reads the data of all pages known to Pico
+     *
+     * @return void
+     */
+    protected function readPages()
+    {
+        $pages = array();
+        $files = $this->getFiles($this->getConfig('content_dir'), $this->getConfig('content_ext'));
+        foreach ($files as $i => $file) {
+            // skip 404 page
+            if (basename($file) == '404' . $this->getConfig('content_ext')) {
+                unset($files[$i]);
+                continue;
+            }
+
+            $id = substr($file, strlen($this->getConfig('content_dir')), -strlen($this->getConfig('content_ext')));
+            $url = $this->getPageUrl($id);
+            if ($file != $this->requestFile) {
+                $rawContent = file_get_contents($file);
+                $meta = $this->parseFileMeta($rawContent, $this->getMetaHeaders());
+            } else {
+                $rawContent = &$this->rawContent;
+                $meta = &$this->meta;
+            }
+
+            // build page data
+            // title, description, author and date are assumed to be pretty basic data
+            // everything else is accessible through $page['meta']
+            $page = array(
+                'id' => $id,
+                'url' => $url,
+                'title' => &$meta['title'],
+                'description' => &$meta['description'],
+                'author' => &$meta['author'],
+                'time' => &$meta['time'],
+                'date' => &$meta['date'],
+                'date_formatted' => &$meta['date_formatted'],
+                'raw_content' => &$rawContent,
+                'meta' => &$meta
+            );
+
+            if ($file == $this->requestFile) {
+                $page['content'] = &$this->content;
+            }
+
+            unset($rawContent, $meta);
+
+            // trigger event
+            $this->triggerEvent('onSinglePageLoaded', array(&$page));
+
+            $pages[$id] = $page;
+        }
+
+        // sort pages by date
+        // Pico::getFiles() already sorts alphabetically
```

Comment:
Like in Pico 0.9, files are strictly sorted in alphabetical order, so `sub/bar.md` and `sub/foo.md` are listed before `sub/index.md`. What do you think about making an exception for index files, so `sub/index.md` is sorted by `sub` instead?